pkg/machinery/config/generate/worker.go

```diff
@@ -96,10 +96,14 @@ func (in *Input) worker() ([]config.Document, error) {
 		machine.MachineKubelet.KubeletDisableManifestsDirectory = pointer.To(true)
 	}
 
-	if in.Options.VersionContract.LocalDNSEnabled() {
+	if in.Options.VersionContract.HostDNSEnabled() || in.Options.HostDNSForwardKubeDNSToHost.ValueOrZero() {
 		machine.MachineFeatures.HostDNSSupport = &v1alpha1.HostDNSConfig{
 			HostDNSEnabled: pointer.To(true),
 		}
+
+		if in.Options.HostDNSForwardKubeDNSToHost.ValueOrZero() {
+			machine.MachineFeatures.HostDNSSupport.HostDNSForwardKubeDNSToHost = pointer.To(true)
+		}
```

Comment:
I think we can set the value directly in `HostDNSSupport`:

```go
		machine.MachineFeatures.HostDNSSupport = &v1alpha1.HostDNSConfig{
			HostDNSEnabled: pointer.To(true),
			HostDNSForwardKubeDNSToHost: in.Options.HostDNSForwardKubeDNSToHost.Ptr(),
		}
```

Reply:
yep, great idea!